daq/host.py

```diff
@@ -386,23 +386,28 @@ def _ping_test(self, src, dst, src_addr=None):
         return self.runner.ping_test(src, dst, src_addr=src_addr)
 
     def _startup_scan(self):
-        assert not self._tcp_monitor, 'tcp_monitor already active'
         self._startup_file = os.path.join(self.scan_base, 'startup.pcap')
         self._startup_time = datetime.now()
         LOGGER.info('Target port %d startup pcap capture', self.target_port)
+        self._tcpdump_scan(self._startup_file)
+
+    def _tcpdump_scan(self, output_file, timeout=None):
+        assert not self._monitor_ref, 'tcp_monitor already active'
         network = self.runner.network
         tcp_filter = ''
-        LOGGER.debug('Target port %d startup scan intf %s filter %s output in %s',
-                     self.target_port, self._mirror_intf_name, tcp_filter, self._startup_file)
+        LOGGER.info('Target port %d scan intf %s for %s filter %s output in %s',
+                     self.target_port, self._mirror_intf_name, timeout, tcp_filter, output_file)
         helper = tcpdump_helper.TcpdumpHelper(network.pri, tcp_filter, packets=None,
                                               intf_name=self._mirror_intf_name,
-                                              timeout=None, pcap_out=self._startup_file,
+                                              timeout=timeout, pcap_out=output_file,
                                               blocking=False)
-        self._tcp_monitor = helper
-        hangup = lambda: self._monitor_error(Exception('startup scan hangup'))
-        self.runner.monitor_stream('tcpdump', self._tcp_monitor.stream(),
-                                   self._tcp_monitor.next_line,
+        self._monitor_ref = helper
+        hangup = self._monitor_complete if timeout else (
```

Comment:
what if scan hangs up when there's a timeout? I think in your hangup function, if there's a timeout, you can check if timeout seconds has elapsed before calling monitor_complete.